h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -222,6 +222,19 @@ public AutoBuffer( ) {
     _persist = 0;               // No persistance
   }
 
+  /** Write to an ever-expanding byte[].  Instead of calling {@link #close()},
+   *  call {@link #buf()} to retrieve the final byte[].  */
+  public AutoBuffer(boolean persist) {
```

Comment:
what is the reason for not using AutoBuffer( OutputStream os, boolean persist )?

Reply:
when I tried to use it, the buffer was overflowing and as a result I got only the end of binary file (only the content which was written to it after overflow)

Reply:
@koniecsveta I think we do need to make the approach outlined by @honzasterba work if there is an issue - I can help with fixing that